### ISSUE TITLE
Can not train a BERT fine tuning model using TF on TPU

### ISSUE THREAD

**Issue by wmmxk** [tensorflow/tensorflow#36194]
I am training a tensorflow model defined as follows on a TPU. But when it starts training, I run into this error: 
UnimplementedError: {{function_node __inference_distributed_function_299068}} Compilation failure: Dynamic concatenation is not supported yet: %reshape.3993 = s32[832]{0} reshape(s32[2,416]{1,0} %reshape.621), metadata={op_type="Reshape" op_name="Reshape_1636"}
	TPU compilation failed
	 [[{{node tpu_compile_succeeded_assert/_14468581000393706631/_10}}]].

I searched on google but did not find any clue to solve this. Any ideas?

def create_model(pretrained_path):

    q_id = tf.keras.layers.Input((MAX_SEQUENCE_LENGTH,), dtype=tf.int32)
    a_id = tf.keras.layers.Input((MAX_SEQUENCE_LENGTH,), dtype=tf.int32)
    q_mask = tf.keras.layers.Input((MAX_SEQUENCE_LENGTH,), dtype=tf.int32)
    a_mask = tf.keras.layers.Input((MAX_SEQUENCE_LENGTH,), dtype=tf.int32)

    q_atn = tf.keras.layers.Input((MAX_SEQUENCE_LENGTH,), dtype=tf.int32)
    a_atn = tf.keras.layers.Input((MAX_SEQUENCE_LENGTH,), dtype=tf.int32)
    config = BertConfig() 

    config.output_hidden_states = False  
    bert_model = TFBertModel.from_pretrained(pretrained_path, config=config)
    q_embedding = bert_model(q_id, attention_mask=q_mask, token_type_ids=q_atn)[0]
    a_embedding = bert_model(a_id, attention_mask=a_mask, token_type_ids=a_atn)[0]
    q = tf.keras.layers.GlobalAveragePooling1D()(q_embedding)
    a = tf.keras.layers.GlobalAveragePooling1D()(a_embedding)

    x =  tf.keras.layers.Concatenate()([q, a])

    x = tf.keras.layers.Dropout(0.2)(x)

    x = tf.keras.layers.Dense(30, activation='sigmoid')(x)

    model = tf.keras.models.Model(inputs=[q_id, q_mask, q_atn, a_id, a_mask, a_atn, ], outputs=x)
    return model

**Comment by gadagashwini** [comment]
@wmmxk, Can you provide the complete code to reproduce the reported issue and also Tensorflow version. Thanks!

**Comment by nariman9119** [comment]
Having same issue.
When starting training with 
```with tpu_strategy.scope():```

It starts to process the first batch and gives me this error

**Comment by gadagashwini** [comment]
@nariman9119 , Can you provide the complete code to replicate the reported issue. Thanks!

**Comment by gadagashwini** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by mikeortman** [comment]
In case anyone comes across this ticket, and frustrated that there was no answer before it was closed, I ran into the exact same issue. For me, my issue was my TF Dataset pipeline causing this issue downstream. 

it was caused by TPUs lack of support for dynamic batch sizes. In TF2, this can be resolved by adding drop_remainder=True to your dataset pipeline's batch operation:

`dataset.batch(..., drop_remainder=True)`

Hope this helps!!
